app/assets/javascripts/discourse/app/lib/sidebar/user/community-section/my-drafts-section-link.js

```diff
@@ -2,35 +2,25 @@ import { tracked } from "@glimmer/tracking";
 import BaseSectionLink from "discourse/lib/sidebar/base-community-section-link";
 import { i18n } from "discourse-i18n";
 
-const USER_DRAFTS_CHANGED_EVENT = "user-drafts:changed";
+const DRAFTS_CHANGED_EVENT = "user-drafts:changed";
 
 export default class MyDraftsSectionLink extends BaseSectionLink {
-  @tracked draftCount = this.currentUser?.draft_count;
+  @tracked shouldDisplay = this._hasDraft;
```

Comment:
Not sure why this isn't working, but all this indirection with appEvents is pretty convoluted. How about simplifying to


```
get shouldDisplay(){
  return this.currentUser?.get("draft_count");
}
```
Any template which references that will be automatically re-rendered when `draft_count` property on the user model is changed.

Note, we have to use `.get()` here for autotracking to work, because `draft_count` is not an `@tracked` property on the user model. If you prefer, you could go and add the `@tracked`, and then simply do `this.currentUser?.draft_count`.